build-tools/src/main/java/co/elastic/apm/compile/tools/publishing/ApmPublisherRootPlugin.java

```diff
@@ -35,12 +37,29 @@ public void apply(Project project) {
     private void configureVersion(Project project) {
         String versionOverride = getVersionOverride(project);
         if (versionOverride != null) {
+            validateVersionOverrideFormatting(versionOverride);
+            validateVersionPatchChangeOnly(project.getVersion().toString(), versionOverride);
             System.out.println("Overriding version with: '" + versionOverride + "'");
             project.setVersion(versionOverride);
             project.subprojects(subproject -> subproject.setVersion(versionOverride));
         }
     }
 
+    private static void validateVersionPatchChangeOnly(String currentVersion, String versionOverride) {
+        VersionNumber comparableVersion = VersionNumber.parse(currentVersion);
+        VersionNumber comparableVersionOverride = VersionNumber.parse(versionOverride);
+        if (comparableVersionOverride.getMajor() > comparableVersion.getMajor() || comparableVersionOverride.getMinor() > comparableVersion.getMinor()) {
+            throw new IllegalArgumentException(String.format("The version override, '%s', cannot provide greater major or minor numbers than the existing version from the gradle.properties file: '%s'.", versionOverride, currentVersion));
```

Comment:
Can you explain what you're checking here? Is it making sure that the new version's minor number is not greater than the existing version's minor number? In that case, how do you release a new minor?

Reply:
> Can you explain what you're checking here? Is it making sure that the new version's minor number is not greater than the existing version's minor number?

Yes, checking that neither the major nor the minor version numbers are greater than the project's next version, which is located [here](https://github.com/elastic/apm-agent-android/blob/main/gradle.properties#L5). This code only runs when releasing a patch version, in which case we would manually pass the patch version in the release Github action parameters which will override the existing project's version. However, since it's a free-format text right now, I'm adding this validation to make sure we won't accidentally release a new minor or major version number.

> how do you release a new minor?

The normal release process is unchanged in this PR, currently, the next minor version is the one defined as the project's version [here](https://github.com/elastic/apm-agent-android/blob/main/gradle.properties#L5). So when we run a normal release, without overriding the project's next version, the project's version will get used in the generated artifact. After the release succeeds, we automatically create a PR, [like this one](https://github.com/elastic/apm-agent-android/pull/277), where we bump the minor version for the next release.